test_utils/prefixer.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import random
+import re
+
+
+_RESOURCE_DATE_FORMAT = "%Y%m%d%H%M%S"
+_RESOURCE_DATE_LENGTH = 4 + 2 + 2 + 2 + 2 + 2
+_RE_SEPARATORS = re.compile(r"[/\-\\_]")
+
+
+def _common_prefix(repo, relative_dir, separator="_"):
+    repo = re.sub(_RE_SEPARATORS, separator, repo)
+    relative_dir = re.sub(_RE_SEPARATORS, separator, relative_dir)
```

Comment:
Given that we compile the pattern at import time, this is clearer:

```suggestion
    repo = _RE_SEPARATORS.sub(separator, repo)
    relative_dir = _RE_SEPARATORS.sub(separator, relative_dir)
```